### ISSUE TITLE
Inconsistency: dns.resolve()

### ISSUE THREAD

**Issue by ameshkov** [hober/web-extensions#570]
This inconsistency was discussed during the WECG in-person meeting.

Both Chrome and Firefox support API for using the local DNS resolver.

* https://developer.chrome.com/docs/extensions/reference/api/dns#method-resolve
* https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/dns/resolve

The Firefox version provides more and we would really like to have it supported in Chrome.

How it can be used quickly once it's implemented:
* Functionality for blocking (or discovering) CNAME-trackers
* Service discovery

What needs to be done before that:
* Chrome devs said that the API currently is only available in the dev channel only due to some security concerns. It is important to learn what exactly are these concerns.
* Firefox and Chrome provide slightly different interfaces. Both APIs are already in use. How can this be handled?

**Comment by oliverdunk** [comment]
Switching from needs-triage to follow-up. On the Chrome side we are going to reach out to our security team to understand if the browser.dns API could be safely exposed.

**Comment by Rob--W** [comment]
Assigned to Oliver to follow up on the Chrome part; tentatively putting https://github.com/w3c/webextensions/labels/supportive%3A%20firefox on this because the requested capabilities are in Firefox, and if there are differences, we're supportive of seeking alignment.

**Comment by kzar** [comment]
@oliverdunk I noticed [this Chromium issue](https://issues.chromium.org/issues/40123151) just got closed out as WONTFIX. I'm not sure what the solution here is, but FWIW this is something I ran into in the past as well. Reading the old discussion, [this comment](https://issues.chromium.org/issues/40123151#comment9) got me thinking in particular, since I wonder if the security concerns would still apply if the `declarativeNetRequest` API allowed extensions to block/redirect CNAME aliased requests somehow. If we could figure it out, that would avoid extension authors needing to add long lists of aliased domains in their `declarativeNetRequest` rules (and to keep those lists updated as shady websites use new random domains).

Edit: In fact, [this comment ](https://issues.chromium.org/issues/40123151#comment11) even said that was planned:
> We also now plan to support this for extensions as part of declarativeNetRequest API.
